pymc3/distributions/simulator.py

```diff
@@ -13,27 +13,77 @@
 #   limitations under the License.
 
 import numpy as np
-from .distribution import NoDistribution
+from .distribution import NoDistribution, draw_values
 
 __all__ = ["Simulator"]
 
 
 class Simulator(NoDistribution):
-    def __init__(self, function, *args, params=None, **kwargs):
+    def __init__(
+        self,
+        function,
+        *args,
+        params=None,
+        distance="gaussian_kernel",
+        sum_stat="identity",
+        epsilon=1,
+        **kwargs,
+    ):
         """
         This class stores a function defined by the user in python language.
         
         function: function
             Simulation function defined by the user.
         params: list
             Parameters passed to function.
+        distance: str or callable
+            Distance functions. Available options are "gaussian_kernel" (default), "wasserstein",
+            "energy" or a user defined function
+            ``gaussian_kernel`` :math: `\sum \left(-0.5  \left(\frac{xo - xs}{\epsilon}\right)^2\right)`
+            ``wasserstein`` :math: `\frac{1}{n} \sum{\left(\frac{|xo - xs|}{\epsilon}\right)}`
+            ``energy`` :math: `\sqrt{2} \sqrt{\frac{1}{n} \sum \left(\frac{|xo - xs|}{\epsilon}\right)^2}`
+            For the wasserstein and energy distances the observed data xo and simulated data xs
+            are internally sorted (i.e. the sum_stat is "sort").
+        sum_stat: str or callable
+            Summary statistics. Available options are ``indentity``, ``sort``, ``mean``, ``median``.
+            If a callable is based it should return a number or a 1d numpy array.
+        epsilon: float
+            Standard deviation of the gaussian_kernel.
         *args and **kwargs: 
             Arguments and keywords arguments that the function takes.
         """
 
         self.function = function
         self.params = params
         observed = self.data
+        self.epsilon = epsilon
+
+        if distance == "gaussian_kernel":
+            self.distance = gaussian_kernel
+        elif distance == "wasserstein":
+            self.distance = wasserstein
+            sum_stat = "sort"
```

Comment:
Can we do a assertion here or a warning that `sum_stat` need to be `"sort"` for `wasserstein` and `energy`?